src/coreclr/nativeaot/Runtime/arm64/WriteBarriers.S

```diff
@@ -22,29 +22,29 @@
         //  $refReg: objectref to be stored
         //
         // On exit:
-        //  x9,x10: trashed
+        //  x12,x17: trashed
         //  other registers are preserved
         //
         .macro UPDATE_GC_SHADOW destReg, refReg
 
         // If g_GCShadow is 0, don't perform the check.
-        PREPARE_EXTERNAL_VAR_INDIRECT g_GCShadow, X9
-        cbz     x9, 1f
+        PREPARE_EXTERNAL_VAR_INDIRECT g_GCShadow, X12
```

Comment:
CoreCLR write barriers seem to be using more efficient pattern to access the statics on arm64. Do you happen to know how much does it save?

Reply:
Are you referring to the version that patches constants in the code? 
I think accessing inline constants is a bit more compact, but on arm64 it is not as nice as on x64 as you can't just load an immediate value, and CPU is likely trained to recognize the indirect load pattern.
Otherwise, these are values that change extremely rarely, so they would be in every cache. In terms of memory fetches it is the same, but fewer instructions.  

We could implement the scheme with patching inline constants and it is likely to be an improvement, but likely too small to consider it an urgency.